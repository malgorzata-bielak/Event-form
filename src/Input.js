import React from "react";

const Input = ({ labelName, id, event }) => (
  <label htmlFor={id}>
    {labelName}
    <input id={id} onChange={event} />
  </label>
);

export default Input;
