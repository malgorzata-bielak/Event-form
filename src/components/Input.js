import React from "react";
import styled from "@emotion/styled";
import { css } from "@emotion/core";

export const Label = styled.label`
  display: flex;
  flex-direction: column;
  font-size: 16px;
  margin: 0;
  color: #2273cf;
`;

export const textField = css`
  border: 1px solid #cacccd;
  height: 35px;
  margin: 5px 0 15px;
  padding-left: 8px;
  outline: none;
  font-size: 16px;
  color: #333;
`;

const InputField = styled.input`
  ${textField}
`;

const Input = ({ labelName, id, event, autofocus }) => (
  <>
    <Label htmlFor={id}>
      {labelName}
      <InputField id={id} onChange={event} autoFocus={autofocus} required />
    </Label>
  </>
);

export default Input;
