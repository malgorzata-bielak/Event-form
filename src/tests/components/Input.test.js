import React from "react";
import { shallow } from "enzyme";

import Input from "../../components/Input";

const propsData = [
  {
    labelName: "nameOne",
    id: "userIdOne",
    event: jest.fn()
  },
  {
    labelName: "nameTwo",
    id: "userIdTwo",
    event: jest.fn(),
    autofocus: true
  }
];

test("should render Input correctly", () => {
  const wrapper = shallow(<Input {...propsData[0]} />);
  expect(wrapper).toMatchSnapshot();
});

test("check if input have autofocus prop", () => {
  const wrapper = shallow(<Input {...propsData[1]} />);
  const input = wrapper.find(`#${propsData[1].id}`);
  expect(input.prop("autoFocus")).toBe(true);
});

test("should set value on input change", () => {
  const event = {
    target: { value: "input value" }
  };
  const wrapper = shallow(<Input {...propsData[1]} />);
  wrapper.find(`#${propsData[1].id}`).simulate("change", event);
  expect(propsData[1].event).toHaveBeenCalledWith(event);
});
