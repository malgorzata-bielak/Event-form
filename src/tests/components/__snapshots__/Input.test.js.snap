// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Input correctly 1`] = `
<Fragment>
  <Styled(label)
    htmlFor="userIdOne"
  >
    name
    <Styled(input)
      id="userIdOne"
      onChange={[MockFunction]}
      required={true}
    />
  </Styled(label)>
</Fragment>
`;
